### ISSUE TITLE
Error with Optional Annotated types

### ISSUE THREAD

**Issue by yinchi** [drivendataorg/erdantic#122]
The following code:
```py
from typing import Optional

import erdantic
from annotated_types import Annotated
from bson import ObjectId
from pydantic import AfterValidator, BaseModel


def assert_object_id(s: str):
    """Assert that a given string is a valid BSON ObjectId and return the string."""
    assert ObjectId.is_valid(s)
    return s

IdStr = Annotated[str, AfterValidator(assert_object_id)]

class Test1(BaseModel):
    id_str: IdStr

class Test2(BaseModel):
    id_str: Optional[IdStr]

erdantic.draw(Test1, out='test1.png')
erdantic.draw(Test2, out='test2.png')
```

results in the following error:
```
RuntimeWarning: Error: not well-formed (invalid token) in line 1 
... Optional[Annotated[str, AfterValidator(func=<function assert_object_id at 0x7fe7076beac0> ...
in label of node __main__.Test2
```

I am on erdantic 1.0.3, annotated_types 0.7.0, and pydantic 2.8.2.

**Comment by jayqi** [comment]
This is indeed a bug. Currently, erdantic special-cases the outermost `Annotated` to strip extra metadata. This doesn't kick in if the `Annotated` is wrapped by anything.

I think the right way to solve this is in the typenames dependency library that we use for turning raw type annotations into string representations. I just made this associated issue: https://github.com/jayqi/typenames/issues/7

**Comment by jayqi** [comment]
This has been released in v1.0.4.
